kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -151,6 +154,21 @@ private void updateStubs() {
     try {
       List<SearchResult<LogMessage>> searchResults =
           searchFuture.get(READ_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+      // we want to be super sure the inner LOG.debug statements are not getting evaluated and then
+      // thrown away because debug is enabled in production as this is a hot code path
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("searchResults.size() = " + searchResults.size());
+        for (SearchResult<LogMessage> searchResult : searchResults) {
+          LOG.debug(
```

Comment:
Instead of doing a loop can't we just print the list directly? The toString methods have reasonable defaults.